app/controllers/cloud_object_store_container_controller.rb

```diff
@@ -32,8 +37,97 @@ def self.display_methods
     %w(cloud_object_store_objects)
   end
 
+  def new
+    assert_privileges("cloud_object_store_container_new")
+    @in_a_form = true
+    if params[:storage_manager_id]
+      @storage_manager = find_record_with_rbac(ExtManagementSystem, params[:storage_manager_id])
+    end
+    @provider_regions = retrieve_provider_regions
+    drop_breadcrumb(
+      :name => _("Add New %{model}") % {:model => ui_lookup(:table => 'cloud_object_store_container')},
+      :url  => "/cloud_object_store_container/new"
+    )
+  end
+
+  def create
+    assert_privileges("cloud_object_store_container_new")
+    case params[:button]
+    when "cancel"
+      javascript_redirect :action    => 'show_list',
+                          :flash_msg => _("Add of new %{model} was cancelled by the user") %
+                                        {:model => ui_lookup(:table => 'cloud_object_store_container')}
+    when "add"
+      options = form_params_create
+      ext_management_system = options.delete(:ems)
+
+      # Queue task
+      task_id = CloudObjectStoreContainer.cloud_object_store_container_create_queue(
+        session[:userid],
+        ext_management_system,
+        options
+      )
+
+      if task_id.kind_of?(Integer)
+        initiate_wait_for_task(:task_id => task_id, :action => "create_finished")
+      else
+        add_flash(_("Cloud Object Store Container creation failed: Task start failed"), :error)
+        javascript_flash(:spinner_off => true)
+      end
+    end
+  end
+
+  def create_finished
+    task_id = session[:async][:params][:task_id]
+    container_name = session[:async][:params][:name]
+    task = MiqTask.find(task_id)
+    if MiqTask.status_ok?(task.status)
+      add_flash(_("Cloud Object Store Container \"%{name}\" created") % {
+        :name => container_name
+      })
+    else
+      add_flash(_("Unable to create Cloud Object Store Container \"%{name}\": %{details}") % {
+        :name    => container_name,
+        :details => task.message
+      }, :error)
+    end
+
+    @breadcrumbs.pop if @breadcrumbs
+    session[:flash_msgs] = @flash_array.dup if @flash_array
+
+    javascript_redirect :action => "show_list"
```

Comment:
Can you change this to :
```ruby
javascript_redirect previous_breadcrumb_url
```

 Useful while adding a New Cloud Object Store Container from the Storage Manager Summary page. 
Once the Add is done, the page will redirected back to the Summary page.